### ISSUE TITLE
Uncaught Error: Operands could not be broadcast together with shapes 1,128,128,64 and 0.

### ISSUE THREAD

**Issue by kanchenhao** [tensorflow/tfjs#938]
#### Code to reproduce the bug

> Uncaught Error: Operands could not be broadcast together with shapes 1,128,128,64 and 0.
    at assertAndGetBroadcastShape (tf-core.esm.js:17)
    at new BatchNormProgram (tf-core.esm.js:17)
    at e.batchNormalization (tf-core.esm.js:17)
    at ENV.engine.runKernel.$x (tf-core.esm.js:17)
    at tf-core.esm.js:17
    at e.scopedRun (tf-core.esm.js:17)
    at e.runKernel (tf-core.esm.js:17)
    at batchNormalization_ (tf-core.esm.js:17)
    at batchNormalization (tf-core.esm.js:17)
    at executeOp$11 (tf-converter.esm.js:17)

**Comment by pyu10055** [comment]
@kanchenhao can you try the latest tfjs-converter python pip package? it should rewrite the batch norm op with basic ops.

**Comment by kanchenhao** [comment]
> @kanchenhao can you try the latest tfjs-converter python pip package? it should rewrite the batch norm op with basic ops.

I had tried the latest tfjs-converter version 0.6.7, but the error is still here.

**Comment by pyu10055** [comment]
@kanchenhao can you share your model, so we can reproduce this problem? thanks.

**Comment by kanchenhao** [comment]
@pyu10055 em...yeah, thank you. You can see the project here [pix2pix-with-tfjs](https://github.com/kanchenhao/pix2pix-with-tfjs), and the Frozenmodel.pb is here [FrozenModel.pb](https://github.com/kanchenhao/pix2pix-with-tfjs/blob/master/FrozenModel.pb). Hope you can help me a lot. Thanks again!

**Comment by pyu10055** [comment]
@nsthorat Can you take a look at the batchNormalization gpu implementation, his model has
mean and variance in the shape of [0], basically both are empty tensors.
The op failed on broadcast check https://github.com/tensorflow/tfjs-core/blob/master/src/kernels/webgl/batchnorm_gpu.ts#L32.

**Comment by kanchenhao** [comment]
@pyu10055 Yes, the frozen model is saved after training. And it's ok when I use the model in python.

**Comment by nsthorat** [comment]
Wait, how could this tensor have no shape? Are you sure the model works in Python?

**Comment by aweinmann** [comment]
Hi, 
I'm having a similar problem with the following model: https://github.com/ychfan/tf_estimator_barebone/files/2502414/wdsr.zip

Inference in python works fine.

Converting also succeeds (tensorflowjs_converter --input_format=tf_saved_model --output_node_names=clip_by_value --saved_model_tags=serve . /web_model)

when predicting in js I get the following error:

> Uncaught (in promise) Error: Operands could not be broadcast together with shapes 3 and 1,64,512,24.
    at assertAndGetBroadcastShape (broadcast_util.ts:94)
    at add_ (binary_ops.ts:61)
    at add (operation.ts:46)
    at executeOp (arithmetic_executor.ts:32)
    at executeOp$16 (operation_executor.ts:51)
    at graph_executor.ts:168
    at engine.ts:156
    at e.scopedRun (engine.ts:167)
    at e.tidy (engine.ts:153)
    at e.tidy (environment.ts:186)

**Comment by allenwangcheng** [comment]
> @pyu10055 Yes, the frozen model is saved after training. And it's ok when I use the model in python.

Have you solve this problem?

**Comment by allenwangcheng** [comment]
@kanchenhao have you solved this problem? I have the same problem with you.

**Comment by rthadur** [comment]
@kanchenhao @pyu10055 gentle ping if we can close this issue ?

**Comment by rthadur** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
